butane_cli/src/lib.rs

```diff
@@ -259,10 +261,46 @@ pub fn get_migrations() -> Result<FsMigrations> {
     Ok(migrations::from_root(root))
 }
 
+/// The members of a workspace are given a PackageId, which contains the directory
+/// name however it needs to be extracted.
+pub fn extract_package_directory(
+    package_id: cargo_metadata::PackageId,
+) -> Result<std::path::PathBuf> {
+    let mut file_url = package_id
+        .repr
+        .split_once(&"path+".to_string())
+        .unwrap()
+        .1
+        .to_string();
+    file_url.truncate(file_url.len() - 1);
+    let url = Url::parse(file_url.as_str()).expect("Failed to parse URL");
+    let file_path = url.to_file_path().unwrap();
+    Ok(file_path)
+}
+
+/// Find a .butane directory to act as the base for butane.
 pub fn base_dir() -> Result<PathBuf> {
-    std::env::current_dir()
-        .map(|d| d.join(".butane"))
-        .map_err(|e| e.into())
+    let current_directory = std::env::current_dir()?;
+    let local_butane_dir = current_directory.join(".butane/");
+
+    if !local_butane_dir.exists() {
+        let metadata = MetadataCommand::new().no_deps().exec()?;
```

Comment:
We should handle the error here rather than just propagating it. If I try to run `butane` in a directory with no Cargo project/workspace today, it prints "No Butane connection info found. Did you run butane init?". That message could arguably be improved, but it at least takes a stab at explaining the problem and offering a mitigation.

With this change it prints only "Encountered unexpected error: `cargo metadata` exited with an error: error: could not find `Cargo.toml` in `/home/james/projects/mood` or any parent directory"